### DISCUSSION TITLE
ControlNet unable to install insightface

### DISCUSSION THREAD

**Discussion by CrinstalMaiden** [lshqqytiger/stable-diffusion-webui-directml#420] [Q&A]
Hi,

It's me again. I just reverted the version from 1.8 back to 1.7.0. Sorry for bothering but I cannot install controlnet properly through the WebUI. Here is the error message:

> Installing sd-webui-controlnet requirement: insightface████████████████████████████████| 23/23 [00:28<00:00,  1.15s/it]
Couldn't install sd-webui-controlnet requirement: insightface.
Command: "E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe" -m pip install https://github.com/Gourieff/Assets/raw/main/Insightface/insightface-0.7.3-cp310-cp310-win_amd64.whl --prefer-binary
Error code: 1
stdout: Collecting insightface==0.7.3
  Using cached https://github.com/Gourieff/Assets/raw/main/Insightface/insightface-0.7.3-cp310-cp310-win_amd64.whl (841 kB)
Collecting scikit-learn
  Using cached scikit_learn-1.4.1.post1-cp310-cp310-win_amd64.whl (10.6 MB)
Requirement already satisfied: Pillow in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (9.5.0)
Requirement already satisfied: scikit-image in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (0.21.0)
Requirement already satisfied: easydict in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (1.13)
Collecting cython
  Using cached Cython-3.0.9-cp310-cp310-win_amd64.whl (2.8 MB)
Requirement already satisfied: numpy in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (1.23.5)
Requirement already satisfied: scipy in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (1.12.0)
Requirement already satisfied: tqdm in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (4.66.2)
Requirement already satisfied: requests in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (2.31.0)
Collecting albumentations
  Using cached albumentations-1.4.1-py3-none-any.whl (130 kB)
Requirement already satisfied: matplotlib in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (3.8.3)
Requirement already satisfied: prettytable in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (3.10.0)
Requirement already satisfied: onnx in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from insightface==0.7.3) (1.15.0)
Collecting opencv-python-headless>=4.9.0
  Using cached opencv_python_headless-4.9.0.80-cp37-abi3-win_amd64.whl (38.5 MB)
Collecting numpy
  Using cached numpy-1.26.4-cp310-cp310-win_amd64.whl (15.8 MB)
Requirement already satisfied: typing-extensions>=4.9.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from albumentations->insightface==0.7.3) (4.10.0)
Requirement already satisfied: PyYAML in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from albumentations->insightface==0.7.3) (6.0.1)
Requirement already satisfied: lazy_loader>=0.2 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (0.3)
Requirement already satisfied: packaging>=21 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (24.0)
Requirement already satisfied: tifffile>=2022.8.12 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (2024.2.12)
Requirement already satisfied: networkx>=2.8 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (3.2.1)
Requirement already satisfied: imageio>=2.27 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (2.34.0)
Requirement already satisfied: PyWavelets>=1.1.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-image->insightface==0.7.3) (1.5.0)
Requirement already satisfied: threadpoolctl>=2.0.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from scikit-learn->insightface==0.7.3) (3.3.0)
Collecting joblib>=1.2.0
  Using cached joblib-1.3.2-py3-none-any.whl (302 kB)
Requirement already satisfied: fonttools>=4.22.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (4.50.0)
Requirement already satisfied: pyparsing>=2.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (3.1.2)
Requirement already satisfied: kiwisolver>=1.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (1.4.5)
Requirement already satisfied: python-dateutil>=2.7 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (2.9.0.post0)
Requirement already satisfied: contourpy>=1.0.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (1.2.0)
Requirement already satisfied: cycler>=0.10 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->insightface==0.7.3) (0.12.1)
Requirement already satisfied: protobuf>=3.20.2 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from onnx->insightface==0.7.3) (3.20.3)
Requirement already satisfied: wcwidth in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from prettytable->insightface==0.7.3) (0.2.13)
Requirement already satisfied: charset-normalizer<4,>=2 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from requests->insightface==0.7.3) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from requests->insightface==0.7.3) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from requests->insightface==0.7.3) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from requests->insightface==0.7.3) (2024.2.2)
Requirement already satisfied: colorama in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from tqdm->insightface==0.7.3) (0.4.6)
Requirement already satisfied: six>=1.5 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from python-dateutil>=2.7->matplotlib->insightface==0.7.3) (1.16.0)
Installing collected packages: numpy, joblib, cython, opencv-python-headless, scikit-learn, albumentations, insightface
  Attempting uninstall: numpy
    Found existing installation: numpy 1.23.5
    Uninstalling numpy-1.23.5:
      Successfully uninstalled numpy-1.23.5
stderr: WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
    WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'E:\\Stable Diffusion\\stable-diffusion-webui-directml\\venv\\Lib\\site-packages\\~1mpy\\.libs\\libopenblas.FB5AE2TYXYH2IJRDKGDGQ3XBKLKTF43H.gfortran-win_amd64.dll'
Check the permissions.
WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages)
[notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe -m pip install --upgrade pip
ControlNet init warning: Unable to install insightface automatically. 

Please help me to fix this error. I have zero knowledge on coding. Thank you so much.

**Comment by lshqqytiger** [comment]
Remove venv folder and try again

**Reply by CrinstalMaiden** [reply]
I did give it a try. And this time the error message is like this:

> Installing sd-webui-controlnet requirement: fvcore
Installing sd-webui-controlnet requirement: mediapipe
Couldn't install sd-webui-controlnet requirement: mediapipe.
Command: "E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe" -m pip install mediapipe --prefer-binary
Error code: 1
stdout: Collecting mediapipe
  Using cached mediapipe-0.10.11-cp310-cp310-win_amd64.whl (50.8 MB)
Collecting opencv-contrib-python
  Using cached opencv_contrib_python-4.9.0.80-cp37-abi3-win_amd64.whl (45.3 MB)
Requirement already satisfied: matplotlib in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (3.8.3)
Requirement already satisfied: flatbuffers>=2.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (24.3.7)
Requirement already satisfied: protobuf<4,>=3.11 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (3.20.3)
Requirement already satisfied: absl-py in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (2.1.0)
Collecting jax
  Using cached jax-0.4.25-py3-none-any.whl (1.8 MB)
Requirement already satisfied: numpy in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (1.23.5)
Requirement already satisfied: attrs>=19.1.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe) (23.2.0)
Collecting sounddevice>=0.4.4
  Using cached sounddevice-0.4.6-py3-none-win_amd64.whl (199 kB)
Collecting CFFI>=1.0
  Using cached cffi-1.16.0-cp310-cp310-win_amd64.whl (181 kB)
Collecting ml-dtypes>=0.2.0
  Using cached ml_dtypes-0.3.2-cp310-cp310-win_amd64.whl (127 kB)
Requirement already satisfied: scipy>=1.9 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from jax->mediapipe) (1.12.0)
Collecting opt-einsum
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Requirement already satisfied: pillow>=8 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (9.5.0)
Requirement already satisfied: kiwisolver>=1.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (1.4.5)
Requirement already satisfied: cycler>=0.10 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (0.12.1)
Requirement already satisfied: packaging>=20.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (24.0)
Requirement already satisfied: pyparsing>=2.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (3.1.2)
Requirement already satisfied: python-dateutil>=2.7 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (2.9.0.post0)
Requirement already satisfied: contourpy>=1.0.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (1.2.0)
Requirement already satisfied: fonttools>=4.22.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe) (4.50.0)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Requirement already satisfied: six>=1.5 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from python-dateutil>=2.7->matplotlib->mediapipe) (1.16.0)
Installing collected packages: pycparser, opt-einsum, opencv-contrib-python, ml-dtypes, jax, CFFI, sounddevice, mediapipe

> stderr: ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'E:\\Stable Diffusion\\stable-diffusion-webui-directml\\venv\\Lib\\site-packages\\cv2\\cv2.pyd'
Check the permissions.


> [notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe -m pip install --upgrade pip

> ControlNet init warning: Unable to install insightface automatically.
Installing sd-webui-controlnet requirement: handrefinerportable
Couldn't install sd-webui-controlnet requirement: handrefinerportable.
Command: "E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe" -m pip install -U https://github.com/huchenlei/HandRefinerPortable/releases/download/v1.0.1/handrefinerportable-2024.2.12.0-py2.py3-none-any.whl --prefer-binary
Error code: 1
stdout: Collecting handrefinerportable==2024.2.12.0
  Downloading https://github.com/huchenlei/HandRefinerPortable/releases/download/v1.0.1/handrefinerportable-2024.2.12.0-py2.py3-none-any.whl (13.1 MB)
     --------------------------------------- 13.1/13.1 MB 38.4 MB/s eta 0:00:00
Collecting mediapipe
  Using cached mediapipe-0.10.11-cp310-cp310-win_amd64.whl (50.8 MB)
Collecting trimesh[easy]
  Using cached trimesh-4.2.0-py3-none-any.whl (689 kB)
Collecting rtree
  Using cached Rtree-1.2.0-py3-none-win_amd64.whl (481 kB)
Requirement already satisfied: absl-py in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (2.1.0)
Requirement already satisfied: protobuf<4,>=3.11 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (3.20.3)
Requirement already satisfied: flatbuffers>=2.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (24.3.7)
Collecting jax
  Using cached jax-0.4.25-py3-none-any.whl (1.8 MB)
Requirement already satisfied: numpy in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (1.26.4)
Collecting opencv-contrib-python
  Using cached opencv_contrib_python-4.9.0.80-cp37-abi3-win_amd64.whl (45.3 MB)
Requirement already satisfied: attrs>=19.1.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (23.2.0)
Requirement already satisfied: matplotlib in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from mediapipe->handrefinerportable==2024.2.12.0) (3.8.3)
Collecting sounddevice>=0.4.4
  Using cached sounddevice-0.4.6-py3-none-win_amd64.whl (199 kB)
Requirement already satisfied: lxml in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (5.1.0)
Collecting shapely
  Using cached shapely-2.0.3-cp310-cp310-win_amd64.whl (1.4 MB)
Requirement already satisfied: pillow in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (9.5.0)
Requirement already satisfied: xxhash in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (3.4.1)
Requirement already satisfied: jsonschema in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (4.21.1)
Collecting mapbox-earcut
  Using cached mapbox_earcut-1.0.1-cp310-cp310-win_amd64.whl (71 kB)
Collecting xatlas
  Using cached xatlas-0.0.9-cp310-cp310-win_amd64.whl (193 kB)
Requirement already satisfied: chardet in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (5.2.0)
Requirement already satisfied: networkx in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (3.2.1)
Requirement already satisfied: httpx in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (0.24.1)
Requirement already satisfied: scipy in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (1.12.0)
Collecting pycollada
  Using cached pycollada-0.8.tar.gz (108 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting vhacdx
  Using cached vhacdx-0.0.6-cp310-cp310-win_amd64.whl (127 kB)
Requirement already satisfied: setuptools in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (63.2.0)
Collecting svg.path
  Using cached svg.path-6.3-py2.py3-none-any.whl (16 kB)
Requirement already satisfied: colorlog in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from trimesh[easy]->handrefinerportable==2024.2.12.0) (6.8.2)
Collecting embreex
  Using cached embreex-2.17.7.post4-cp310-cp310-win_amd64.whl (13.4 MB)
Collecting CFFI>=1.0
  Using cached cffi-1.16.0-cp310-cp310-win_amd64.whl (181 kB)
Requirement already satisfied: colorama in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from colorlog->trimesh[easy]->handrefinerportable==2024.2.12.0) (0.4.6)
Requirement already satisfied: certifi in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (2024.2.2)
Requirement already satisfied: httpcore<0.18.0,>=0.15.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (0.15.0)
Requirement already satisfied: sniffio in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (1.3.1)
Requirement already satisfied: idna in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (3.6)
Requirement already satisfied: opt-einsum in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from jax->mediapipe->handrefinerportable==2024.2.12.0) (3.3.0)
Collecting ml-dtypes>=0.2.0
  Using cached ml_dtypes-0.3.2-cp310-cp310-win_amd64.whl (127 kB)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from jsonschema->trimesh[easy]->handrefinerportable==2024.2.12.0) (2023.12.1)
Requirement already satisfied: rpds-py>=0.7.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from jsonschema->trimesh[easy]->handrefinerportable==2024.2.12.0) (0.18.0)
Requirement already satisfied: referencing>=0.28.4 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from jsonschema->trimesh[easy]->handrefinerportable==2024.2.12.0) (0.34.0)
Requirement already satisfied: python-dateutil>=2.7 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (2.9.0.post0)
Requirement already satisfied: cycler>=0.10 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (4.50.0)
Requirement already satisfied: pyparsing>=2.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (3.1.2)
Requirement already satisfied: kiwisolver>=1.3.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (1.4.5)
Requirement already satisfied: contourpy>=1.0.1 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (1.2.0)
Requirement already satisfied: packaging>=20.0 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (24.0)
Requirement already satisfied: pycparser in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from CFFI>=1.0->sounddevice>=0.4.4->mediapipe->handrefinerportable==2024.2.12.0) (2.21)
Requirement already satisfied: h11<0.13,>=0.11 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpcore<0.18.0,>=0.15.0->httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (0.12.0)
Requirement already satisfied: anyio==3.* in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from httpcore<0.18.0,>=0.15.0->httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (3.7.1)
Requirement already satisfied: exceptiongroup in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from anyio==3.*->httpcore<0.18.0,>=0.15.0->httpx->trimesh[easy]->handrefinerportable==2024.2.12.0) (1.2.0)
Requirement already satisfied: six>=1.5 in e:\stable diffusion\stable-diffusion-webui-directml\venv\lib\site-packages (from python-dateutil>=2.7->matplotlib->mediapipe->handrefinerportable==2024.2.12.0) (1.16.0)
Using legacy 'setup.py install' for pycollada, since package 'wheel' is not installed.
Installing collected packages: xatlas, vhacdx, trimesh, svg.path, shapely, rtree, opencv-contrib-python, ml-dtypes, mapbox-earcut, embreex, CFFI, sounddevice, pycollada, jax, mediapipe, handrefinerportable

> stderr: ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'E:\\Stable Diffusion\\stable-diffusion-webui-directml\\venv\\Lib\\site-packages\\cv2\\cv2.pyd'
Check the permissions.

> [notice] A new release of pip available: 22.2.1 -> 24.0
[notice] To update, run: E:\Stable Diffusion\stable-diffusion-webui-directml\venv\Scripts\python.exe -m pip install --upgrade pip

> Warning: Failed to install handrefinerportable. Some processors will not work.
Installing sd-webui-controlnet requirement: depth_anything

And the installation stop right here.

**Reply by CrinstalMaiden** [reply]
I just updated the error message. Looks like not only the insightface, but also other components of the controlnet.

**Reply by lshqqytiger** [reply]
It doesn't seem to be a problem of stable-diffusion-webui-directml. Maybe a compatibility issue?

**Reply by CrinstalMaiden** [reply]
So what else can I try to get controlnet work? Is there any workaround I can do to install controlnet successfully? It worked before Dec 2023 update, so I don't think compatibility is the cause here T_T

**Reply by CrinstalMaiden** [reply]
I've tried to revert back to 1.6.1 but the problem is still there. Should I try to update pip to see if that work?

**Reply by lshqqytiger** [reply]
You'd better to search on the repository of the extension: https://github.com/Mikubill/sd-webui-controlnet/issues/2297
